src/main/java/com/mycompany/mszczepienia/service/VisitService.java

```diff
@@ -112,6 +115,21 @@ public void updateMissedVisitsStatus() {
         });
     }
 
+    @Transactional
+    public VisitDto cancelVisit(Long visitId){
+        System.out.println(visitId);
+        Visit visit = visitRepository.getById(visitId);
+        System.out.println(visit);
+        visit.setVisitStatus(VisitStatus.CANCELLED);
```

Comment:
When a visit is made, we decrement vaccine quantity by 1, when a visit is cancelled or missed we should increment it back (see method updateMissedVisitsStatus()). Simply add provided line:
``
placeVaccineRepository.incrementQuantity(visit.getPlace().getId(), visit.getVaccine().getId());
``